### PR TITLE
feat(pipelines): add feature to display all changesets for stage before manual approval step + added postPrepare Step

### PR DESCRIPTION
**Pull request by Nico-DB**
Describe the feature

I would like to "run" all changesets of a stage at first, then have a manual approval step and then perform the execute change set in the original run-order.

Use Case

The use case is, that we would like to check the changesets for all stacks of an environment, to be able to perform a qualified manual approval.

Closes https://github.com/aws/aws-cdk/issues/24072.
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk-testing/framework-integ/test/pipelines/test/integ.newpipeline-with-allPrepareNodesFirst.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+    const pipeline = new pipelines.CodePipeline(this, 'Pipeline', {
+      synth: new pipelines.ShellStep('Synth', {
+        input: pipelines.CodePipelineSource.gitHub('Nico-DB/aws-cdk', 'main'),
+        commands: ['npm ci', 'npm run build', 'npx cdk synth'],
+      }),
+      allPrepareNodesFirst: true,
+    });
+
+
```
Comment (scanlonp) [inline packages/@aws-cdk-testing/framework-integ/test/pipelines/test/integ.newpipeline-with-allPrepareNodesFirst.ts:21]: Build is yelling at you here for having multiple blank lines.

File: packages/@aws-cdk-testing/framework-integ/test/pipelines/test/integ.newpipeline-with-vpc.ts [hunk 2]
```diff
@@ -1,14 +1,16 @@
...
 import * as s3_assets from 'aws-cdk-lib/aws-s3-assets';
 import * as sqs from 'aws-cdk-lib/aws-sqs';
-import { App, Stack, StackProps, Stage, StageProps } from 'aws-cdk-lib';
-import { Construct } from 'constructs';
 import * as pipelines from 'aws-cdk-lib/pipelines';
+import { Construct } from 'constructs';
+import * as path from 'path';
 import { PIPELINE_REDUCE_ASSET_ROLE_TRUST_SCOPE } from 'aws-cdk-lib/cx-api';
 
+
```
Comment (scanlonp) [inline packages/@aws-cdk-testing/framework-integ/test/pipelines/test/integ.newpipeline-with-vpc.ts:13]: Build is yelling at you here for having multiple blank lines.

Comment (Nico-DB) [reply packages/@aws-cdk-testing/framework-integ/test/pipelines/test/integ.newpipeline-with-vpc.ts]: has been resolved

File: packages/@aws-cdk-testing/framework-integ/test/pipelines/test/integ.newpipeline-with-allPrepareNodesFirst.ts (hunk 1 above)
Comment (Nico-DB) [reply packages/@aws-cdk-testing/framework-integ/test/pipelines/test/integ.newpipeline-with-allPrepareNodesFirst.ts]: has been resolved

**Comment by scanlonp** [comment]
Hey @Nico-DB, it looks like the build is failing with some more serious errors. It appears that some tests are failing with destructive changes. We generally cannot give a review until the build is passing. Also if these destructive changes are necessary or intended, we will need a strong justification.

**Comment by Nico-DB** [comment]
Hey @scanlonp,
thanks for checking the errors. Could you explain what destructive changes means here? Since i am trying to add this feature for more then a year now, it could be that things changed in the meantime. When i created this feature and testet, i could not see destructive changes

**Pull request closed without merging**